homeassistant/components/template/cover.py

```diff
@@ -141,57 +140,56 @@ class CoverTemplate(TemplateEntity, CoverEntity):
 
     def __init__(
         self,
-        hass,
+        hass: HomeAssistant,
         object_id,
-        config,
+        config: dict[str, Any],
         unique_id,
-    ):
+    ) -> None:
         """Initialize the Template cover."""
         super().__init__(
             hass, config=config, fallback_name=object_id, unique_id=unique_id
         )
         self.entity_id = async_generate_entity_id(
             ENTITY_ID_FORMAT, object_id, hass=hass
         )
-        friendly_name = self._attr_name
+        name = self._attr_name
+        if TYPE_CHECKING:
+            assert name is not None
         self._template = config.get(CONF_VALUE_TEMPLATE)
         self._position_template = config.get(CONF_POSITION_TEMPLATE)
         self._tilt_template = config.get(CONF_TILT_TEMPLATE)
         self._attr_device_class = config.get(CONF_DEVICE_CLASS)
-        self._open_script = None
-        if (open_action := config.get(OPEN_ACTION)) is not None:
-            self._open_script = Script(hass, open_action, friendly_name, DOMAIN)
-        self._close_script = None
-        if (close_action := config.get(CLOSE_ACTION)) is not None:
-            self._close_script = Script(hass, close_action, friendly_name, DOMAIN)
-        self._stop_script = None
-        if (stop_action := config.get(STOP_ACTION)) is not None:
-            self._stop_script = Script(hass, stop_action, friendly_name, DOMAIN)
-        self._position_script = None
-        if (position_action := config.get(POSITION_ACTION)) is not None:
-            self._position_script = Script(hass, position_action, friendly_name, DOMAIN)
-        self._tilt_script = None
-        if (tilt_action := config.get(TILT_ACTION)) is not None:
-            self._tilt_script = Script(hass, tilt_action, friendly_name, DOMAIN)
+
+        if (open_config := config.get(OPEN_ACTION)) is not None:
+            self.add_script(hass, OPEN_ACTION, open_config, name, DOMAIN)
+
+        if (close_config := config.get(CLOSE_ACTION)) is not None:
+            self.add_script(hass, CLOSE_ACTION, close_config, name, DOMAIN)
+
+        # The config requires open and close scripts, therefore the base
+        # supported features will always include them.
```

Comment:
This comment is lying, cover config requires *either* open+close scripts *or* a set position script, which is why open and close is always supported

Reply:
Ah yes, I missed the cv.has_at_least_one_key at the bottom of the schema